### PR TITLE
start files for full test on tower via aws batch

### PR DESCRIPTION
**Pull request by taylorreiter**
I think the profile template conf is here: https://github.com/nf-core/configs/blob/master/conf/aws_tower.config

The goal of this PR is to set up full tests that will run automatically when there's a new release. Through secrets I set up on the repo, this should automagically launch on our tower account and dump the results into the seqqc s3 bucket. The test data sets are in [Arcadia-Science/test-datasets](https://github.com/Arcadia-Science/test-datasets/tree/main/seqqc) already, and I have already tested them on Tower and they run with the current `dev` branch of seqqc. Doing all of these chemistries will also mean that we produce example multiqc reports every time we make a new release, which I think is a good idea.

**The only thing I'm not sure about is whether each data set should have its own workflow.** It made sense to me that it would, but i'm not sure.

### CODE REVIEW

File: .github/workflows/towerfulltest.yml
Comment (mertcelebi) [inline .github/workflows/towerfulltest.yml:14]: should we pin these to a specific version?

File: .github/workflows/towerfulltest.yml
Comment (mertcelebi) [inline .github/workflows/towerfulltest.yml:1]: I'd rename these files to `tower_full_test_*`

File: .github/workflows/towerfulltest_hifi.yml
Comment (mertcelebi) [inline .github/workflows/towerfulltest_hifi.yml:1]: So much repeat code :/ I wonder if you could parameterize these.

File: .github/workflows/towerfulltest_illumina.yml
Comment (mertcelebi) [inline .github/workflows/towerfulltest_illumina.yml:27]: what does the aws_tower profile do?

File: .github/workflows/towerfulltest.yml
Comment (mertcelebi) [inline .github/workflows/towerfulltest.yml]:
```suggestion
              "outdir" : "s3://${{ secrets.AWS_S3_BUCKET }}/full-tests/results-${{ github.sha }}/full"
```

**Review by mertcelebi — COMMENTED** [review]
LGTM, but curious to see if we can dedupe some of the scaffold code to reduce duplication

File: .github/workflows/towerfulltest.yml
Comment (taylorreiter) [reply .github/workflows/towerfulltest.yml:14]: I don't feel super strongly about this. I kind of like that it will force us to keep things current on releases, even if that does cause some dependency issues. Those are things we should probably be aware of and address anyway so, 🤷‍♀️ . Happy to pin if you feel strongly about it

File: .github/workflows/towerfulltest_hifi.yml
Comment (taylorreiter) [reply .github/workflows/towerfulltest_hifi.yml:1]: Ya i felt that when I was writing this. I looked at nf-core to see if other people ran multiple AWS full tests to see if there was a better way to structure this and I couldn't find any examples. I'm a relative newbie with github actions/workflows, do you have any ideas of how these could be combined?

File: .github/workflows/towerfulltest_illumina.yml
Comment (taylorreiter) [reply .github/workflows/towerfulltest_illumina.yml:27]: docs: https://github.com/nf-core/configs/blob/master/docs/aws_tower.md
config: https://github.com/nf-core/configs/blob/master/conf/aws_tower.config

File: .github/workflows/towerfulltest.yml
Comment (mertcelebi) [reply .github/workflows/towerfulltest.yml:14]: Sounds good, this is fine

File: .github/workflows/towerfulltest_hifi.yml
Comment (mertcelebi) [reply .github/workflows/towerfulltest_hifi.yml:1]: Okay, so you can do this basically: https://docs.github.com/en/actions/creating-actions/creating-a-composite-action.

But seems annoying and we can potentially get to it later. Doesn't seem super high priority.

**Review by mertcelebi — APPROVED** [review]
I am going to approve this for now, but some comments:
* Feel free to do or not do the action composition stuff to dedupe the actions code. 
* I'd like to also dedupe all these profiles at some point. But none of this seems that essential for now. 
* One thing I noticed is: you probably have to add these new profiles to the `nextflow.config` file. Make sure that change is done before you merge. That file in general may require some updates (DOI, version etc), so noting that here.

**Comment by taylorreiter** [comment]
Wait did I miss something re the action composition stuff? does that comment highlight a strategy or is there something more I should be seeing? 

in either case, I'll take a look at the nextflow config, thank you for catching this!

re DOI/version -- those can't be updated until after the first release when zenodo assigns the pipeline a DOI.

**Comment by mertcelebi** [comment]
@taylorreiter I added a comment about it inline. This is the relevant strategy for deduping: https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

File: .github/workflows/towerfulltest.yml
Comment (taylorreiter) [reply .github/workflows/towerfulltest.yml:1]: I modeled these after what nf-core pipelines use:
"name: nf-core AWS full size tests"
https://github.com/nf-core/taxprofiler/blob/dev/.github/workflows/awsfulltest.yml#L1

File: .github/workflows/towerfulltest_hifi.yml
Comment (taylorreiter) [reply .github/workflows/towerfulltest_hifi.yml:1]: ok posted as issue #34 and will deal with it in the future!

**Pull request merged**
